File: Go/sort/bubble_sort.go
package main

import "fmt"

// Bubble sort, sometimes referred to as sinking sort, 
// is a simple sorting algorithm that repeatedly steps through the list, 
// compares adjacent elements and swaps them if they are in the wrong order.
func bubble_sort(arr []int,num int) []int {
	for i:= 0; i <= num - 1; i++ {
		for j := 0; j < num - i - 1; j++ {
			// Checking if the adjacent elements are not in order.
			if (arr[j] > arr[j + 1]) {
				arr[j], arr[j + 1] = arr[j + 1], arr[j]
			}
		} 
	}

	return arr
}

func main() {
	var size int
	fmt.Printf("Please enter the size of the array: ")
	fmt.Scan(&size)

	var arr = make([]int, size)
	for itr := 0; itr < size; itr++ {
		fmt.Printf("Enter the element %d: ", itr + 1)
		fmt.Scan(&arr[itr])
	}

	fmt.Println("The entered array is:", arr)

	sorted := bubble_sort(arr, size)
	fmt.Println("The sorted array is:", sorted)
}

File: Julia/sort/bubble_sort.jl
"""Julia program to implement Bubble Sort
In Bubble Sort we compare the adjacent elements and swap them such that the greater
element among them comes in the right-side. After each iteration, one smallest element
goes towars the end of the array.
"""

function bubble_sort(arr, n)
    # Iterate the entire array
    for i in 1:n
        # Iterate till the concerned part only.
        for j in 1:(n-i)
            # Swap adjacent elements if they are in wrong order
            if(arr[j] > arr[j+1])
                arr[j], arr[j+1] = arr[j+1], arr[j]
            end
        end
    end
end
            

print("How many numbers do you want to sort? ")
n = readline()
n = parse(Int64, n)
if (n <= 0)
    println("No numbers to sort!!!")
    exit()
end
arr = Int[]
print("Enter the numbers: ")
arr = [parse(Int, num) for num in split(readline())]
bubble_sort(arr, n)
print("The numbers in sorted order is ")
print(arr)



"""
Time Complexity - O(n^2), where 'n' is the size of the array
Space Complexity - O(n)

SAMPLE INPUT AND OUTPUT

SAMPLE I
How many numbers do you want to sort? 5
Enter the numbers: 4 2 5 1 3
The numbers in sorted order is [1, 2, 3, 4, 5]

"""
